rest-service/manager_rest/rest/resources_v3_1/plugins.py

```diff
@@ -70,3 +75,85 @@ def post(self, **kwargs):
         )
         with rest_utils.skip_nested_marshalling():
             return super(Plugins, self).post(visibility=visibility)
+
+
+class PluginsUpdate(SecuredResource):
+
+    @rest_decorators.exceptions_handled
+    @authorize('plugins_update_create')
+    @rest_decorators.marshal_with(models.PluginsUpdate)
+    def post(self, id, phase):
+        """
+        Supports two stages of a plugin update.
+        Phases:
+            1. Creates a temporary blueprint and executes a deployment update
+            (will update only the plugins) for all the deployments of the
+            given blueprint.
+            2. Updates the original blueprint plan and deletes the temp one.
+
+        :param id: the blueprint ID to update it's deployments' plugins if
+        phase == 'initiate', otherwise (phase == 'finalize') the plugin update
+        ID.
+        :param phase: either 'initiate' or 'finalize' (internal).
+        """
+        if phase == PHASES.INITIAL:
```

Comment:
if "phase" is binary, why not use a bool flag?

Reply:
More human friendly IMO, `False/True` won't have any true meaning and I think it'll be confusing especially in case of tracking rest calls

Reply:
It's kind of error-prone (string comp)